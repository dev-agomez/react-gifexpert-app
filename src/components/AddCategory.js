import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => { 
        if (e) {
            setInputValue(e.target.value);
            console.log("hadleInputChange llamado");    

            if(inputValue === ""){
                setCategories(x => [""]);   
            }
        }
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        console.log('handleSubmit', inputValue)
        if (inputValue) {
            //setCategories(x => [inputValue,...x]);  
            setCategories(x => [inputValue]);  
            setInputValue("");  
        }
    }

    const handleKeyPress = (event) => {
        if (inputValue && inputValue.length >= 2) {
            setCategories(x => [inputValue]);   
        }

        if(inputValue === ""){
            setCategories(x => [""]);   
        }
        console.log(inputValue);
      }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onKeyPress={handleKeyPress}
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
}