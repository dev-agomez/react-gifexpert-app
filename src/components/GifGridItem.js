import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({id, title, url, index}) => {
    return (
        <div className="col-md-4 animate__animated animate__lightSpeedInLeft">
            <div className="card mb">
                <a href={url} target="_blank">
                    <img className="card-img-top" src={url} alt={title}/>
                </a>                 
                <div className="card-body mb">
                    <h5 className="card-title">{index + ".- " +  title}</h5>
                </div>
            </div>
        </div>
)}

GifGridItem.propTypes = {
    title : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    index: PropTypes.number.isRequired
}