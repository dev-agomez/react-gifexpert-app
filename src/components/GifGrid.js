import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {
    
    const { data: images,loading, notFound } = useFetchGifs(category);

    return (
       <>
        {/* <h3 className="animate__animated animate__lightSpeedInLeft">{ category }</h3> */}
        {loading && <p className="animate__animated animate__flash">Cargando...</p>}
        {notFound && <p className="animate__animated animate__bounceInDown">No hay elementos para mostrar...</p>}
        {
            <div className="card-grid">
                { 
                    images.map(img => (
                        <GifGridItem 
                        key={img.id} 
                        {...img}
                        />
                    ))
                }
            </div>
        }
       </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}