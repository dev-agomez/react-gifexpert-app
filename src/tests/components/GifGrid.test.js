import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import { GifGrid } from '../../components/GifGrid';

describe('Pruebas en GifGrid', () => {
    const category = 'Goku';
    const wrapper = shallow(<GifGrid category={category}/>);

    test('Debe mostrarse correctamente.', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
