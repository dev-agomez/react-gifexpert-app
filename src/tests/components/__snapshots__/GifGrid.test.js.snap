// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe mostrarse correctamente. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__lightSpeedInLeft"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <p
    className="animate__animated animate__bounceInDown"
  >
    No hay elementos para mostrar...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
